File: core/src/main/java/com/mcreater/amclcore/model/oauth/TokenResponseModel.java
package com.mcreater.amclcore.model.oauth;

import com.google.gson.annotations.SerializedName;
import com.mcreater.amclcore.annotations.OAuthLoginModel;
import com.mcreater.amclcore.annotations.RequestModel;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
@OAuthLoginModel
@RequestModel
public class TokenResponseModel {
    @SerializedName("token_type")
    private String tokenType;
    @SerializedName("expires_in")
    private int expiresIn;
    @SerializedName("ext_expires_in")
    private int extExpiresIn;
    private String scope;
    @SerializedName("access_token")
    private String accessToken;
    @SerializedName("refresh_token")
    private String refreshToken;
    private String error;
    @SerializedName("error_description")
    private String errorDescription;
    @SerializedName("correlation_id")
    private String correlationId;
}

File: core/src/main/java/com/mcreater/amclcore/model/oauth/TokenResponseModel.kt
package com.mcreater.amclcore.model.oauth

import com.google.gson.annotations.SerializedName
import com.mcreater.amclcore.annotations.OAuthLoginModel
import com.mcreater.amclcore.annotations.RequestModel

@RequestModel
@OAuthLoginModel
data class TokenResponseModel(
    @SerializedName("token_type") var tokenType: String? = null,
    @SerializedName("expires_in") var expiresIn: Int = 0,
    @SerializedName("ext_expires_in") var extExpiresIn: Int = 0,
    var scope: String? = null,
    @SerializedName("access_token") var accessToken: String? = null,
    @SerializedName("refresh_token") var refreshToken: String? = null,
    var error: String? = null,
    @SerializedName("error_description") var errorDescription: String? = null,
    @SerializedName("correlation_id") var correlationId: String? = null
)
